lxd/instance/drivers/driver_lxc.go

```diff
@@ -2247,11 +2247,18 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 	_ = unix.Unmount(d.RootfsPath(), unix.MNT_DETACH)
 
 	// Snapshot if needed.
-	err = d.startupSnapshot(d)
+	snapName, expiry, err := d.getSnapNameAndExpiry(d)
 	if err != nil {
 		return "", nil, err
```

Comment:
I know not strictly part of this issue, but lets wrap the error returned here (in both drivers) so its like:

```go
fmt.Errorf("Failed getting startup snapshot info: %w", err)
```